torch_np/_funcs_impl.py

```diff
@@ -1208,6 +1228,68 @@ def outer(a: ArrayLike, b: ArrayLike, out: Optional[OutArray] = None):
     return torch.outer(a, b)
 
 
+def einsum(*operands, out=None, optimize=False, **kwargs):
+    # Have to manually normalize *operands and **kwargs, following the NumPy signature
+    # >>> np.einsum?
+    # Signature: np.einsum(*operands, out=None, optimize=False, **kwargs)
+    # Docstring:
+    # einsum(subscripts, *operands, out=None, dtype=None, order='K',
+    #       casting='safe', optimize=False)
+
+    from ._normalizations import (
+        maybe_copy_to,
+        normalize_casting,
+        normalize_dtype,
+        normalize_not_implemented,
+        normalize_outarray,
+        wrap_tensors,
+    )
```

Comment:
If removing imports is the goal, that won't help really. These normalizers are otherwise only used in `_normalizations.py`, so these need to be imported either way.

Reply:
What I'd like to remove is local imports. They are a massive red flag

Reply:
That would create an import cycle since `_funcs.py` imports `_funcs_impl`.

We can of course, import these at the `_funcs_impl` module level and special-case them to not get dumped into the global namespace, if that's really what you think is needed?

Reply:
We would just need to import things